src/create_disk.sh

```diff
@@ -80,20 +80,55 @@ set -x
 # Partition and create fs's. The 0...4...a...1 uuid is a sentinal used by coreos-gpt-setup
 # in ignition-dracut. It signals that the disk needs to have it's uuid randomized and the
 # backup header moved to the end of the disk.
-sgdisk -Z $disk \
-	-U 00000000-0000-4000-a000-000000000001 \
-	-n 1:0:+384M -c 1:boot \
-	-n 2:0:+127M -c 2:EFI-SYSTEM -t 2:C12A7328-F81F-11D2-BA4B-00A0C93EC93B \
-	-n 3:0:+1M   -c 3:BIOS-BOOT  -t 3:21686148-6449-6E6F-744E-656564454649 \
-	-n 4:0:0     -c 4:root       -t 4:4F68BCE3-E8CD-4DB1-96E7-FBCAF984B709
-sgdisk -p "$disk"
+# Pin /boot and / to the partition number 1 and 4 respectivelly
+BOOTPN=1
+ROOTPN=4
+case "$arch" in
+    x86_64)
+        sgdisk -Z $disk \
+        -U 00000000-0000-4000-a000-000000000001 \
+        -n ${BOOTPN}:0:+384M -c ${BOOTPN}:boot \
+        -n 2:0:+127M -c 2:EFI-SYSTEM -t 2:C12A7328-F81F-11D2-BA4B-00A0C93EC93B \
+        -n 3:0:+1M   -c 3:BIOS-BOOT  -t 3:21686148-6449-6E6F-744E-656564454649 \
+        -n ${ROOTPN}:0:0     -c ${ROOTPN}:root       -t ${ROOTPN}:0FC63DAF-8483-4772-8E79-3D69D8477DE4
+        sgdisk -p "$disk"
+        EFIPN=2
+        BIOSPN=3
+        ;;
+    aarch64)
+        sgdisk -Z $disk \
+        -U 00000000-0000-4000-a000-000000000001 \
+        -n ${BOOTPN}:0:+384M -c ${ROOTPN}:boot \
+        -n 2:0:+127M -c 2:EFI-SYSTEM -t 2:C12A7328-F81F-11D2-BA4B-00A0C93EC93B \
+        -n ${ROOTPN}:0:0     -c ${ROOTPN}:root       -t ${ROOTPN}:0FC63DAF-8483-4772-8E79-3D69D8477DE4
+        sgdisk -p "$disk"
+        EFIPN=2
+        ;;
+    s390x)
+        sgdisk -Z $disk \
+        -U 00000000-0000-4000-a000-000000000001 \
+        -n ${BOOTPN}:0:+384M -c ${BOOTPN}:boot \
+        -n ${ROOTPN}:0:0     -c ${ROOTPN}:root       -t ${ROOTPN}:0FC63DAF-8483-4772-8E79-3D69D8477DE4
+        sgdisk -p "$disk"
+        ;;
+    ppc64le)
+        # ppc64le doesn't use special uuid for root partition
+        sgdisk -Z $disk \
+        -U 00000000-0000-4000-a000-000000000001 \
+        -n 2:0:+4M   -c 2:PowerPC-PReP-boot -t 2:9E1A2D38-C612-4316-AA26-8B49521E5A8B \
+        -n ${BOOTPN}:0:+384M -c ${BOOTPN}:boot \
+        -n ${ROOTPN}:0:0     -c ${ROOTPN}:root              -t ${ROOTPN}:0FC63DAF-8483-4772-8E79-3D69D8477DE4
```

Comment:
Did you mean to leave off the partition type uuid here? Based off of:

>          # ppc64le doesn't use special uuid for root partition 

Or are you just saying that it doesn't help, but it doesn't hurt either?

Reply:
No, this is in the context of https://github.com/coreos/coreos-assembler/issues/795 , If it is too confusing I can drop that.

Reply:
OK gotcha, thanks for the context.
I think we can make this decision more explicit, but let's leave that as a follow-up!